crownheightsaid/mutual-aid-app
# [138] Add Delivery Table fields - Need, Timeline, & Description

Fixes #138 

Add fields from the Slack form to the Delivery Requests Table:
- Need
- Timeline
- Description

Also a little bit of refactoring: moved the Delivery Table Row into its own component and rendering the header row in a function. 

Open to suggestions for ordering of these fields! I tried to match how they are currently ordered in the Slack form.

I decided to label Timeline as "Requested Delivery Time" to convey the difference between its meaning & "Days Open". What do you think? Open to suggestions for this too!

Since Description can be long and the table was already getting pretty wide with the two extra fields, I decided to use the Collapsible Table material UI component. I think it looks pretty nice! I also added a Tool Tip when you hover over the arrow icon which will say show or hide description.

The description will expand whenever the row is in focus. Check it out!

![Aug-19-2020 16-21-43](https://user-images.githubusercontent.com/8590508/90685642-1bc28400-e238-11ea-8ed1-c639a9f0423b.gif)


**To test these changes:**

1. Pull this branch
2. Make sure you're set up to pull from the staging Heroku instance
3. Load `/delivery-requests` page
4. You should see the new fields added and description expanded when you click on a row!



(side note: idk why my git account isn't properly set up to these commits on this computer :sob:)
